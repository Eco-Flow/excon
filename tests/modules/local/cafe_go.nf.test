nextflow_process {

    name "Test Process CAFE_GO"
    script "modules/local/cafe_go.nf"
    process "CAFE_GO"

    test("Should run without failures") {

        when {
            params {
                // define parameters here:
                clean = false
                outdir = "results"
            }
            process {
                """
                // define inputs of the process here.
                input[0] = Channel.of( file("${projectDir}/data/cafe_go_input/N0.tsv") )
                input[1] = Channel.of( file("${projectDir}/data/cafe_go_input/OG_GO_format.tsv") )
                input[2] = Channel.of( file("${projectDir}/data/cafe_go_input/Out_cafe") )
                """
            }
        }

        then {
            //check if test case succeeded
            assert process.success
            //analyze trace file
            assert process.trace.tasks().size() == 1 

            //Assert snapshot of the two main output files
            assert snapshot(
              path("${launchDir}/${params.outdir}/cafe_go/CAFE_summary.txt"),
              path("${launchDir}/${params.outdir}/cafe_go/versions.yml")
            ).match("cafe_go") 
        }
    }
}
