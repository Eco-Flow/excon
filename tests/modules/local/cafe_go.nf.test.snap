{
    "cafe_go": {
        "content": [
            "CAFE_summary.txt:md5,b6cbf4877b159864b97d514273b4ab67",
            "versions.yml:md5,acfc3dae393e700a8add474731579ae8"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-03T12:01:52.31372"
    }
}